advanced/kernel-messaging/README.md

```diff
@@ -159,87 +184,72 @@ export class KernelModel extends VDomModel {
-  ) {
-    this._future = value;
-    if (!value) {
-      return;
-    }
-    value.onIOPub = this._onIOPub;
-  }
-
-  private _output: nbformat.IOutput | null = null;
   private _future: Kernel.IFuture<
     KernelMessage.IExecuteRequestMsg,
     KernelMessage.IExecuteReplyMsg
   > | null = null;
+  private _output: nbformat.IOutput | null = null;
   private _session: IClientSession;
+  private _stateChanged = new Signal<KernelModel, void>(this);
 }
 ```
 
 ## Connecting a View to the Kernel
 
-The only remaining thing left is to connect a View to the Model. We have
-already seen the `TutorialView` before. To trigger the `render` function of a
-`VDomRendered` on a `stateChanged` signal, we just need to add our `VDomModel`
-to `this.model` in the constructor. We can then connect a button to
-`this.model.execute` and a text field to `this.model.output` and our extension
-is ready:
+Now that the session is created and the model is defined, the view can
+be connected to the model to display the execution results.
+
+In this example, the view contains a `UseSignal` React element that listens
+to the `stateChanged` signal defined by the model. Whenever the `stateChanged`
+signal is emitted, the `UseSignal` React element will update its children
+according to the new state of the model. In this example the execution
+results are retrieved through `this._model.output` attribute and display
+in a text field.
 
+<!-- prettier-ignore-start -->
 ```ts
-// src/widget.tsx#L9-L34
+// src/widget.tsx#L26-L30
 
-export class KernelView extends VDomRenderer<any> {
-  constructor(model: KernelModel) {
-    super();
-    this.id = `TutorialVDOM`;
-    this.model = model;
-  }
+<UseSignal signal={this._model.stateChanged}>
+  {() => (
+    <span key="output field">{JSON.stringify(this._model.output)}</span>
+  )}
+</UseSignal>
+```
+<!-- prettier-ignore-end -->
 
-  protected render(): React.ReactElement<any>[] {
-    console.log('render');
-    const elements: React.ReactElement<any>[] = [];
-    elements.push(
-      <button
-        key="header-thread"
-        className="jp-tutorial-button"
-        onClick={() => {
-          this.model.execute('3+5');
-        }}
-      >
-        Compute 3+5
-      </button>,
-
-      <span key="output field">{JSON.stringify(this.model.output)}</span>
-    );
-    return elements;
-  }
-}
+Finally to trigger a statement execution, the click event of a button is
+implemented to call the `this._model.execute` method.
+
+```ts
+// src/widget.tsx#L17-L25
+
+<button
+  key="header-thread"
+  className="jp-example-button"
+  onClick={() => {
+    this._model.execute('3+5');
+  }}
+>
+  Compute 3+5
+</button>
 ```
 
-Well that's nice, the basics are clear, but what about this weird output
-object? In the [Kernel Output](https://github.com/jtpio/jupyterlab-extension-examples/tree/master/advanced/kernel-output)
-example, we will explore how we can reuse some jupyter components to make things look nicer...
+## Where to Go Next
```

Comment:
Having these sections is a really good idea, and very useful to link back to the other examples :+1: